tests/keras/layers/test_advanced_activations.py

```diff
@@ -0,0 +1,160 @@
+import pytest
+from numpy.testing import assert_allclose
+import numpy as np
+from keras import backend as K
+
+
+def get_standard_values():
+    '''
+    These are just a set of floats used for testing the activation
+    functions, and are useful in multiple tests.
+
+    The values should all be non-negative because they and their negatives
+    are used to test the ReLU derivates in this file.
+    '''
+    return np.array([[0, 0.1, 0.5, 0.9, 1.0, 10, 1e2, 0.01]], dtype=K.floatx())
+
+
+def test_leaky_relu():
+    np.random.seed(1337)
+    from keras.layers.advanced_activations import LeakyReLU
+    inp = get_standard_values()
+    for alpha in [0., .5, -1.]:
+        layer = LeakyReLU(alpha=alpha)
+        layer.input = K.variable(inp)
+        for train in [True, False]:
+            outp = K.eval(layer.get_output(train))
+            assert_allclose(outp, inp)
+
+        layer.input = K.variable(-inp)
+        for train in [True, False]:
+            outp = K.eval(layer.get_output(train))
+            assert_allclose(outp, -inp*alpha)
+
+        config = layer.get_config()
+        assert config['alpha'] == alpha
+
+
+def test_prelu():
+    from keras.layers.advanced_activations import PReLU
+    np.random.seed(1337)
+    inp = get_standard_values()
+
+    for train in [True, False]:
+        # test with custom weights
+        alphas = np.random.random(inp.shape)
+        layer = PReLU(weights=alphas, input_shape=inp.flatten().shape)
+        # calling build here causes an error, unclear if this is a bug
```

Comment:
It's looking like these issues will be resolved as soon as TensorFlow supports int -> bool casting (which will also enable masking). We just have to wait.